### PR TITLE
ENH: add sketcher MVP

### PR DESCRIPTION
**Pull request by tacaswell**
Use mouse to draw Line2D objects.

This is a MVP and needs to have many configuration attributes
added (linecolor, linestyle, sketch-mode, smoothing of path etc).

This was written an the request of @fperez 

Does not integrate with the existing widget base classes as I wanted more flexible way of managing multiple call backs.

Usage is 

``` python
fig, ax = plt.subplots()
sk = mpl.widgets.Sketcher(ax)
```

and the every click/drag creates a new `Line2D` artist on the axes.

The lines created are in the `sk.lines` attribute and the most recently added line as `sk.last_line`.

This is the nascent prototype of the idea of building out basic plot manipulation tools.

### CODE REVIEW

**Comment by tacaswell** [comment]
attn @danielballan @ericdill Probably want this for day job as well.

**Comment by danielballan** [comment]
Nice! I often fall back to more GUI-centric tools when I need to draw on my plots. For annotations, drawing by hand is the way to go. It would be able to stay in matplotlib for these tasks.

**Comment by mdboom** [comment]
Combine this with `xkcd()` and you won't be able to tell real data from fake!  :wink:

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -1833,3 +1833,90 @@ def onmove(self, event):
...
+        old_cid = self.cid_cache.pop(msg, None)
+
+        if old_cid is not None:
+            self.ax.figure.canvas.mpl_disconnect(old_cid)
+
+    def _down_event(self, ev):
+        """
+        button down callback
+        """
+        del self.point_cache[:]
```
Comment (WeatherGod) [inline lib/matplotlib/widgets.py]: why clear the cache prior to checking for widgetlock? If the widget doesn't have lock, then it means that the widget wasn't the intended recipient of the signal, right?

**Comment by jklymak** [comment]
If this ever goes somewhere, I'd suggest its an example rather than an MPL feature.  I'll close, but feel free to reopen.

**Pull request closed without merging**